flux-desugar/src/desugar.rs

```diff
@@ -569,8 +598,35 @@ impl<'a, 'tcx> DesugarCtxt<'a, 'tcx> {
                 fhir::TyKind::Array(Box::new(ty), fhir::ArrayLen { val: len.val, span: len.span })
             }
             surface::TyKind::Hole => fhir::TyKind::Hole,
+            surface::TyKind::ImplTrait(res, bounds) => {
+                if let Res::OpaqueTy(def_id) = res &&
+                   let Some(local_def_id) = def_id.as_local()
+                {
+                    let owner_id = OwnerId { def_id: local_def_id };
+                    let item_id = hir::ItemId { owner_id };
+
+                    let arity = self.early_cx.tcx.generics_of(local_def_id).params.len();
+                    let args = Self::mk_generic_args_with_hole(arity, span);
+                    let args = self.desugar_generic_args(*res, &args, binders)?;
```

Comment:
This is still wrong, we have to create `GenericArg`s matching the kind of the corresponding `GenericParam`s.  However, I think these are always guaranteed to be lifetime arguments so we can't just call `desugar_generic_args` with an empty slice and that will insert the appropriate lifetime holes.

```suggestion
                    let args = self.desugar_generic_args(*res, &[], binders)?;
```

Reply:
By **can't just call** you mean **can just call** I presume?

Reply:
ups, yeah, I mean we *can*